storage-operator/pkg/controller/volume/volume_controller.go

```diff
@@ -213,6 +213,7 @@ func newReconciler(mgr manager.Manager) reconcile.Reconciler {
 		scheme:   mgr.GetScheme(),
 		recorder: mgr.GetRecorder("volume-controller"),
 		salt:     salt.NewClient(getAuthCredential(mgr.GetConfig())),
+		saltenv:  "",
```

Comment:
This can't be right, similar how `fetchSaltEnv` can't be right accordingly.

This structure is global to the operator. However, we retrieve a `saltenv` from a `Node`, as intended: the `saltenv` to be used when invoking Salt can be different from one `Volume` to another, when those sit on different `Node`s.

So we'd need something like `getNodeSaltenv` or something (not even cached, as far as I'm concerned), and use the resulting value when invoking Salt, within the scope of handling a single `Volume`.

Reply:
It was working fine because `saltenv` was recomputed at each reconciliation, but it was confusing.
I'll remove the field and use a local variable in the reconciliation loop instead.

Done.